### PR TITLE
DEVX-123 migrate cart discount update action utils

### PR DESCRIPTION
**Pull request by lojzatran**
#### Summary

JIRA: https://commercetools.atlassian.net/browse/DEVX-123

**Boy-scout changes**
I changed the generics namings.

### CODE REVIEW

File: src/main/java/com/commercetools/sync/sdk2/cartdiscounts/utils/CartDiscountUpdateActionUtils.java [hunk 1]
```diff
@@ -0,0 +1,500 @@
...
+
+  @NotNull
+  private static Optional<CartDiscountUpdateAction> buildChangeFixedValueUpdateAction(
+      final CartDiscountValueFixed oldValue, final CartDiscountValueFixedDraft newValue) {
+    return oldValue.getMoney().equals(newValue.getMoney())
+        ? Optional.empty()
+        : Optional.of(CartDiscountChangeValueActionBuilder.of().value(newValue).build());
+  }
+
+  @NotNull
```
Comment (salander85) [inline src/main/java/com/commercetools/sync/sdk2/cartdiscounts/utils/CartDiscountUpdateActionUtils.java:111]:
```suggestion
  @Nonnull
```

File: src/main/java/com/commercetools/sync/sdk2/cartdiscounts/utils/CartDiscountUpdateActionUtils.java [hunk 2]
```diff
@@ -0,0 +1,500 @@
...
+  private static Optional<CartDiscountUpdateAction> buildChangeRelativeValueUpdateAction(
+      final CartDiscountValueRelative oldValue, final CartDiscountValueRelativeDraft newValue) {
```
Comment (salander85) [inline src/main/java/com/commercetools/sync/sdk2/cartdiscounts/utils/CartDiscountUpdateActionUtils.java:113]:
```suggestion
      @Nonnull final CartDiscountValueRelative oldValue, @Nonnull final CartDiscountValueRelativeDraft newValue) {
```

File: src/main/java/com/commercetools/sync/sdk2/cartdiscounts/utils/CartDiscountUpdateActionUtils.java [hunk 3]
```diff
@@ -0,0 +1,500 @@
...
+    return oldValue.getPermyriad().equals(newValue.getPermyriad())
+        ? Optional.empty()
+        : Optional.of(CartDiscountChangeValueActionBuilder.of().value(newValue).build());
+  }
+
+  @NotNull
```
Comment (salander85) [inline src/main/java/com/commercetools/sync/sdk2/cartdiscounts/utils/CartDiscountUpdateActionUtils.java:119]:
```suggestion
  @Nonnull
```

File: src/main/java/com/commercetools/sync/sdk2/cartdiscounts/utils/CartDiscountUpdateActionUtils.java [hunk 4]
```diff
@@ -0,0 +1,500 @@
...
+  private static Optional<CartDiscountUpdateAction> buildChangeGiftLineItemValueUpdateAction(
+      final CartDiscountValueGiftLineItem oldValue,
+      final CartDiscountValueGiftLineItemDraft newValue) {
```
Comment (salander85) [inline src/main/java/com/commercetools/sync/sdk2/cartdiscounts/utils/CartDiscountUpdateActionUtils.java:122]:
```suggestion
      @Nonnull final CartDiscountValueGiftLineItem oldValue,
      @Nonnull final CartDiscountValueGiftLineItemDraft newValue) {
```

**Comment by salander85** [comment]
@lojzatran Please fix the failing code checks and if codeCov decreases try to fix before merging.

File: src/main/java/com/commercetools/sync/sdk2/cartdiscounts/models/CartDiscountCustomTypeAdapter.java [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+
+  /**
+   * Get typeId of the {@link CartDiscount} see:
+   * https://docs.commercetools.com/api/types#referencetype
+   *
+   * @return the typeId "cartDiscount"
+   */
+  @Override
+  public String getTypeId() {
+    return "cartDiscount";
```
[the remark this answers is not in the document]
Comment (lojzatran) [reply src/main/java/com/commercetools/sync/sdk2/cartdiscounts/models/CartDiscountCustomTypeAdapter.java:38]: I could not find `getReferenceTypeId()` method - how did you come up with this?

**Pull request merged**